tests/testthat/test_fragments2.R

```diff
@@ -0,0 +1,144 @@
+## For additional tests, refer to calculateFragments from PSMatch package
+test_that("calculateFragments2: Default behavior without modifications", {
+    
+    ## Test 1: Default behavior without modifications
+    sequence <- "PQR"
+    original_result <- calculateFragments(
+        sequence = sequence,
+        type = c("b", "y"),
+        z = 1,
+        modifications = NULL,
+        neutralLoss = defaultNeutralLoss(),
+        verbose = FALSE
+    )
+    result <- calculateFragments2(
+        sequence = sequence,
+        type = c("b", "y"),
+        z = 1,
+        fixed_modifications = NULL,
+        variable_modifications = NULL,
+        max_mods = Inf,
+        neutralLoss = defaultNeutralLoss(),
+        verbose = FALSE
+    )
+    
+    ## Check unique peptide without modifications
+    expect_equal(unique(result$peptide), "PQR")
+    expect_equal(result[,-7], original_result)
```

Comment:
I would suggest to use names, rather than removing the 7th column. For example:
```
expect_identical(result[, names(original_result)], original_result)
```